apps/condo/domains/miniapp/components/GlobalApps/GlobalAppsContainer.tsx

```diff
@@ -64,6 +75,37 @@ export const GlobalAppsContainer: React.FC = () => {
         }
     }, [])
 
+    const handleTask = useCallback((message) => {
+        const taskRecord = {
+            id: get(message, 'id'),
+            taskId: message.taskId,
+            title: message.taskTitle,
+            progress: message.taskProgress,
+            description: message.taskDescription,
+            status: message.taskStatus,
+            __typename: 'MiniAppTask',
+        }
+
+        if (message.taskOperation === 'create') {
+            taskRecord.id = uuidV4()
+            const createMiniAppTask = miniAppTaskUIInterface.storage.useCreateTask({}, () => {
+                addTask({
+                    ...miniAppTaskUIInterface,
+                    record: taskRecord,
+                })
+            })
+
+            createMiniAppTask(taskRecord)
+
+        } else if (message.taskOperation === 'update') {
+            const updateMiniAppTask = miniAppTaskUIInterface.storage.useUpdateTask({}, () => {
+                updateTask(taskRecord)
+            })
```

Comment:
Could you please uplift this code to component scope, because it is in hooks format, which are not allowed to be initialized in conditions. You can take `taskRecord` from argument of `onComplete`.